src/Diagnostics.HealthChecks/Composables/HealthCheckComposablesExtensions.cs

```diff
@@ -0,0 +1,113 @@
+		this IHealthCheck healthCheck,
+		IMemoryCache cache,
+		ActivitySource activitySource,
+		ILogger logger,
+		HealthCheckParameters? parameters = null)
+	{
+		IHealthCheck startupHealthCheck = new StartupHealthCheck(healthCheck, cache);
+		return new ObservableHealthCheck(parameters ?? new(), startupHealthCheck, activitySource, logger);
+	}
+
+	/// <summary>
+	/// Applies the default error handling and activity around the current health check.
+	/// </summary>
+	/// The instance of the Activity marker. By selecting the implementor, the health check will determine how
+	/// the related endpoint will process the health check request.
+	/// <param name="healthCheck">The health check.</param>
+	/// <param name="activitySource">The activity source.</param>
+	/// <param name="logger">The logger.</param>
+	/// <param name="parameters">The health check parameters, set at DI registration time.</param>
+	/// <returns>The health check with the default configuration and error handling.</returns>
+	public static IHealthCheck AsObservableHealthCheck(
+		this IHealthCheck healthCheck,
+		ActivitySource activitySource,
+		ILogger logger,
+		HealthCheckParameters? parameters = null) =>
+		new ObservableHealthCheck(parameters ?? new(), healthCheck, activitySource, logger);
+
+	/// <summary>
+	/// Creates a new HTTP endpoint checker health check.
+	/// </summary>
+	/// <param name="httpClientFactory">The http client factory.</param>
+	/// <param name="requestBuilder">The request builder.</param>
+	/// <param name="healthCheckResponseChecker">The response checker instance.</param>
+	/// <param name="activitySource">The activity source.</param>
+	/// <param name="httpClientName">The HTTP Client name, if a client with that name was registered in particular.</param>
+	/// <param name="activityMarker">Marks the activity with the required baggage items.</param>
+	/// <returns>The HTTP endpoint checker health check.</returns>
+	public static IHealthCheck CreateHttpHealthCheck(
+		Func<HttpRequestMessage> requestBuilder,
+		Func<HealthCheckContext, HttpResponseMessage, CancellationToken, Task<HealthCheckResult>> healthCheckResponseChecker,
+		ActivitySource activitySource,
+		IHttpClientFactory httpClientFactory,
+		string? httpClientName = "",
+		Action<Activity?>? activityMarker = null) =>
+			new HttpEndpointHealthCheck(
+				httpClientFactory,
+				requestBuilder,
+				healthCheckResponseChecker,
+				activitySource,
+				httpClientName: httpClientName,
+				activityMarker: activityMarker);
+
+	/// <summary>
+	/// Checks whether the response from the endpoint is successful or not by checking its HTTP Status Code.
+	/// </summary>
+	/// <param name="context">The Health Check Context.</param>
+	/// <param name="response">The endpoint response.</param>
+	/// <param name="allowedStatusCodes">The allowed status codes for the response..</param>
+	/// <returns>The health check result.</returns>
+	public static Task<HealthCheckResult> CheckResponseStatusCodeAsync(
```

Comment:
Goes for any other "Async" methods that are using Task.FromResult without async await too

Reply:
That actually surprised me, at this point in time in .NET I think that every time a method returns a `Task`, it's best to simply add `async`, because the overhead is nothing compared to these kind of risks.